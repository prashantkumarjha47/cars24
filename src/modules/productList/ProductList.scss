.product-list {
  display: flex;
  width: 100%;
  padding: 50px 0;
  .product-list-left {
    @media screen and (max-width: 991px) {
      position: fixed;
      left: 0;
      top: 0;
      background-color: #fff;
      box-shadow: 0 0 10px rgba($color: #000000, $alpha: 0.1);
      z-index: 100;
      width: 400px;
      height: 100%;
      transition: all 0.3s ease-in-out;
      transform: scaleX(0);
      transform-origin: left;
    }
    &.open {
      transform: scaleX(1);
    }
  }
  // .product-list-right {
  //   width: 66.66%;
  //   max-width: 66.66%;
  //   flex: 0 0 66.66%;
  // }
  .list {
    border: 1px solid black;
  }
  .sorter-row {
    margin-bottom: 10px;
  }
  .product-row {
    .card-col {
      margin-bottom: 15px;
    }
  }
  .cat {
    display: flex;
    justify-content: flex-end;
  }
  .cart {
    display: flex;
    justify-content: center;
    flex-direction: column;
    max-width: 100%;
    flex-wrap: wrap;
    table {
      border: 1px solid black;
      padding: 1px;
    }
  }
}
