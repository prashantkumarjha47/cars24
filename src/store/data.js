const data = {
  carCopy: [
    {
      id: 1,
      title: "Tata Nexon",
      price: 50000,
      category: "Compact SUV",
      image: "...",
      color: "white",
      condition: "New",
      buildYear: "2020",
      make: "TATA",
      model: "XE",
    },
    /*...*/
    {
      id: 2,
      title: "Kia Sonet",
      price: 225000,
      category: "Compact SUV",
      image: "...",
      color: "black",
      condition: "new",
      buildYear: "2020",
      make: "Kia",
      model: "1.2 HTE 5MT",
    },
    {
      id: 3,
      title: "Ford EcoSport",
      price: 300000,
      category: "Compact SUV",
      image: "...",
      color: "blue",
      condition: "new",
      buildYear: "2020",
      make: "Ford",
      model: "1.5L Petrol Ambiente MT",
    },
    {
      id: 4,
      title: "Hyundai Venue",
      price: 1675000,
      category: "Compact SUV",
      image: "...",
      color: "grey",
      condition: "new",
      buildYear: "2020",
      make: "Hyundai",
      model: "1.2 Kappa MT E",
    },
    {
      id: 5,
      title: "Hyundai Venue",
      price: 875000,
      category: "Compact SUV",
      image: "...",
      color: "grey",
      condition: "new",
      buildYear: "2020",
      make: "Hyundai",
      model: "1.2 Kappa MT E",
    },
    {
      id: 6,
      title: "Hyundai Venue",
      price: 500000,
      category: "Compact SUV",
      image: "...",
      color: "grey",
      condition: "new",
      buildYear: "2020",
      make: "Hyundai",
      model: "1.2 Kappa MT E",
    },
    {
      id: 7,
      title: "Hyundai Venue",
      price: 2575000,
      category: "Compact SUV",
      image: "...",
      color: "grey",
      condition: "new",
      buildYear: "2020",
      make: "Hyundai",
      model: "1.2 Kappa MT E",
    },
    {
      id: 8,
      title: "Hyundai Venue",
      price: 600000,
      category: "Compact SUV",
      image: "...",
      color: "grey",
      condition: "new",
      buildYear: "2020",
      make: "Hyundai",
      model: "1.2 Kappa MT E",
    },
    {
      id: 9,
      title: "Hyundai Venue",
      price: 150000,
      category: "Compact SUV",
      image: "...",
      color: "grey",
      condition: "new",
      buildYear: "2020",
      make: "Hyundai",
      model: "1.2 Kappa MT E",
    },
    {
      id: 10,
      title: "Hyundai Venue",
      price: 480000,
      category: "Compact SUV",
      image: "...",
      color: "grey",
      condition: "new",
      buildYear: "2020",
      make: "Hyundai",
      model: "1.2 Kappa MT E",
    },
    {
      id: 11,
      title: "Hyundai Venue",
      price: 955000,
      category: "Compact SUV",
      image: "...",
      color: "grey",
      condition: "new",
      buildYear: "2020",
      make: "Hyundai",
      model: "1.2 Kappa MT E",
    },
    {
      id: 12,
      title: "Hyundai Venue",
      price: 740000,
      category: "Compact SUV",
      image: "...",
      color: "grey",
      condition: "new",
      buildYear: "2020",
      make: "Hyundai",
      model: "1.2 Kappa MT E",
    },
    {
      id: 13,
      title: "Hyundai Venue",
      price: 520000,
      category: "Premium hatchback",
      image: "...",
      color: "grey",
      condition: "new",
      buildYear: "2020",
      make: "Hyundai",
      model: "1.2 Kappa MT E",
    },
    {
      id: 15,
      title: "Hyundai Venue",
      price: 98000,
      category: "Premium hatchback",
      image: "...",
      color: "grey",
      condition: "new",
      buildYear: "2020",
      make: "Hyundai",
      model: "1.2 Kappa MT E",
    },
    {
      id: 16,
      title: "Hyundai Venue",
      price: 375000,
      category: "Hatchback",
      image: "...",
      color: "grey",
      condition: "new",
      buildYear: "2020",
      make: "Hyundai",
      model: "1.2 Kappa MT E",
    },
    {
      id: 17,
      title: "Hyundai Venue",
      price: 347000,
      category: "Hatchback",
      image: "...",
      color: "grey",
      condition: "new",
      buildYear: "2020",
      make: "Hyundai",
      model: "1.2 Kappa MT E",
    },
    {
      id: 18,
      title: "Hyundai Venue",
      price: 552000,
      category: "Hatchback",
      image: "...",
      color: "grey",
      condition: "new",
      buildYear: "2020",
      make: "Hyundai",
      model: "1.2 Kappa MT E",
    },
    {
      id: 19,
      title: "Hyundai Venue",
      price: 614000,
      category: "Hatchback",
      image: "...",
      color: "grey",
      condition: "new",
      buildYear: "2020",
      make: "Hyundai",
      model: "1.2 Kappa MT E",
    },
    {
      id: 20,
      title: "Hyundai Venue",
      price: 1005000,
      category: "Hatchback",
      image: "...",
      color: "grey",
      condition: "new",
      buildYear: "2020",
      make: "Hyundai",
      model: "1.2 Kappa MT E",
    },
    {
      id: 21,
      title: "Hyundai Venue",
      price: 75000,
      category: "Hatchback",
      image: "...",
      color: "grey",
      condition: "new",
      buildYear: "2020",
      make: "Hyundai",
      model: "1.2 Kappa MT E",
    },
  ],
  categories: [
    {
      id: 0,
      title: "",
    },
    {
      id: 1,
      title: "Compact SUV",
    },
    {
      id: 2,
      title: "Premium hatchback",
    },
    {
      id: 3,
      title: "Hatchback",
    },
  ],
  condition: [
    {
      id: 0,
      title: "",
    },
    {
      id: 1,
      type: "used",
    },
    {
      id: 2,
      type: "damaged",
    },
  ],
  makes: [
    {
      id: 0,
      title: "",
    },
    {
      id: 1,
      name: "TATA",
    },
    {
      id: 2,
      name: "Kia",
    },
    {
      id: 3,
      name: "Hyundai",
    },
    {
      id: 4,
      name: "Ford",
    },
    {
      id: 5,
      name: "Maruti",
    },
  ],
};

export default data;
