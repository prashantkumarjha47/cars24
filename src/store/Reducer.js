import data from "./data";

const OFFSET = 4;
const arr = data.carCopy.sort((a, b) => a.price - b.price);
const initialState = {
  ...data,
  n: OFFSET,
  cars: data.carCopy.slice(0, OFFSET),
  minVal: arr[0].price,
  maxValu: arr[arr.length - 1].price,
  cart: [],
};

const reducer = (state, action) => {
  switch (action.type) {
    case "GET_CARS":
      const { category, priceRange, vehicleCondition, make, search } = state;
      let filteredCars = state.carCopy
        .filter((car) => {
          return (
            (!category || car.category === category) &&
            (!priceRange || car.price <= priceRange) &&
            (!vehicleCondition || car.vehicleCondition === vehicleCondition) &&
            (!make || car.make === make) &&
            (!search || car.title.toLowerCase().includes(search.toLowerCase()))
          );
        })
        .slice(0, state.n + OFFSET);

      if (state.sort) {
        filteredCars = filteredCars.sort((a, b) =>
          state.sort === "ASC"
            ? parseFloat(a.price) - parseFloat(b.price)
            : parseFloat(b.price) - parseFloat(a.price)
        );
      }

      return {
        ...state,
        n: state.n + OFFSET,
        cars: filteredCars,
      };

    case "GET_CATEGORIES":
      return {
        ...state,
        categories: [...state.categories],
      };
    case "GET_CONDITION":
      return {
        ...state,
        condition: [...state.condition],
      };
    case "SEARCH_CAR":
      return {
        ...state,
        search: action.payload,
        cars: state.cars.filter((car) =>
          car.title?.toLowerCase().includes(action.payload?.toLowerCase())
        ),
      };
    case "SORT_CAR_ASC":
      return {
        ...state,
        sort: "ASC",
        cars: state.cars.sort(
          (a, b) => parseFloat(a.price) - parseFloat(b.price)
        ),
      };
    case "SORT_CAR_DESC":
      return {
        ...state,
        sort: "DESC",
        cars: state.cars.sort(
          (a, b) => parseFloat(b.price) - parseFloat(a.price)
        ),
      };
    case "APPLY_FILTER":
      return {
        ...state,
        n: OFFSET,
        ...action.payload,
        cars: [],
      };
    case "BOOK_NOW":
      return {
        ...state,
        cart: [...state.cart, action.payload],
      };

    default:
      return state;
  }
};

export { reducer, initialState };
