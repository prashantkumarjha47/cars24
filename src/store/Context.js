import React, { useReducer } from "react";
import { reducer, initialState } from "./Reducer";

const Context = React.createContext();

const ContextProvider = (props) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <Context.Provider value={[state, dispatch]}>
      {props.children}
    </Context.Provider>
  );
};

export { ContextProvider, Context };
