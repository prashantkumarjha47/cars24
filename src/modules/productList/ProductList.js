import React, { useContext, useState, useEffect } from "react";
import {
  Card,
  CardImg,
  CardBody,
  CardTitle,
  CardSubtitle,
  Button,
  Container,
  Row,
  Col,
} from "reactstrap";
import { Context } from "../../store/Context";
import Sorter from "../../components/sorter/Sorter";
import Filters from "../../components/filter/Filter";
import "./ProductList.scss";

export default function ProductList() {
  const [state, dispatch] = useContext(Context);
  const [drawer, setDrawer] = useState(false);
  const [isBottom, setIsBottom] = useState(false);

  useEffect(() => {
    window.addEventListener("scroll", handleScroll);
    return () => window.removeEventListener("scroll", handleScroll);
  }, []);

  useEffect(() => {
    if (isBottom) {
      dispatch({ type: "GET_CARS" });
      setIsBottom(false);
    }
  }, [isBottom, dispatch]);

  const handleScroll = () => {
    const scrollTop =
      (document.documentElement && document.documentElement.scrollTop) ||
      document.body.scrollTop;
    const scrollHeight =
      (document.documentElement && document.documentElement.scrollHeight) ||
      document.body.scrollHeight;
    if (scrollTop + window.innerHeight + 10 >= scrollHeight) {
      setIsBottom(true);
    }
  };

  const onFilterClick = () => {
    setDrawer(!drawer);
  };

  const addCarToCart = (car) => {
    dispatch({ type: "BOOK_NOW", payload: car });
  };

  return (
    <div className="product-list">
      <div className="container">
        <Row>
          <Col
            xs="12"
            sm="12"
            md="12"
            lg="4"
            xl="4"
            className={`product-list-left ${drawer ? "open" : ""}`}
          >
            <Filters onCloseClick={onFilterClick} />
          </Col>
          <Col
            xs="12"
            sm="12"
            md="12"
            lg="8"
            xl="8"
            className="product-list-right"
          >
            <div className="sorter-row">
              <Sorter onFilterClick={onFilterClick} />
            </div>
            <div className="product-row">
              <Container>
                <Row>
                  {state.cars.map((car, i) => {
                    if (car.category === "Hatchback") {
                      return (
                        <Col
                          xs="12"
                          sm="6"
                          md="4"
                          lg="4"
                          xl="4"
                          key={i}
                          className="card-col"
                        >
                          <Card>
                            <div className="cat">
                              <h5>*</h5>
                            </div>
                            <CardImg
                              top
                              width="100%"
                              src="https://stimg.cardekho.com/images/car-images/large/Tata/Nexon/7297/1579769288048/224_Flame-Red_a52411.jpg"
                              alt="Card image cap"
                            />
                            <CardBody>
                              <CardTitle tag="h5">{car.title + " "}</CardTitle>
                              <CardSubtitle
                                tag="h6"
                                className="mb-2 text-muted"
                              >
                                {`₹ ${car.price}`}
                              </CardSubtitle>
                              <Button onClick={() => addCarToCart(car)}>
                                Book Now
                              </Button>
                            </CardBody>
                          </Card>
                        </Col>
                      );
                    } else if (car.category === "Premium hatchback") {
                      return (
                        <Col
                          xs="12"
                          sm="6"
                          md="4"
                          lg="4"
                          xl="4"
                          key={i}
                          className="card-col"
                        >
                          <Card>
                            <div className="cat">
                              <h5>$</h5>
                            </div>
                            <CardImg
                              top
                              width="100%"
                              src="https://stimg.cardekho.com/images/car-images/large/Tata/Nexon/7297/1579769288048/224_Flame-Red_a52411.jpg"
                              alt="Card image cap"
                            />
                            <CardBody>
                              <CardTitle tag="h5">{car.title + " "}</CardTitle>
                              <CardSubtitle
                                tag="h6"
                                className="mb-2 text-muted"
                              >
                                {`₹ ${car.price}`}
                              </CardSubtitle>
                              <Button onClick={() => addCarToCart(car)}>
                                Book Now
                              </Button>
                            </CardBody>
                          </Card>
                        </Col>
                      );
                    } else {
                      return (
                        <Col
                          xs="12"
                          sm="6"
                          md="4"
                          lg="4"
                          xl="4"
                          key={i}
                          className="card-col"
                        >
                          <Card>
                            <div className="cat">
                              <h5>#</h5>
                            </div>
                            <CardImg
                              top
                              width="100%"
                              src="https://stimg.cardekho.com/images/car-images/large/Tata/Nexon/7297/1579769288048/224_Flame-Red_a52411.jpg"
                              alt="Card image cap"
                            />
                            <CardBody>
                              <CardTitle tag="h5">{car.title + " "}</CardTitle>
                              <CardSubtitle
                                tag="h6"
                                className="mb-2 text-muted"
                              >
                                {`₹ ${car.price}`}
                              </CardSubtitle>
                              <Button onClick={() => addCarToCart(car)}>
                                Book Now
                              </Button>
                            </CardBody>
                          </Card>
                        </Col>
                      );
                    }
                  })}
                </Row>
              </Container>
            </div>
          </Col>
        </Row>
        <div>
          <div className="cart">
            <div>
              <h4>Cart</h4>
            </div>
            <table>
              <tr>
                <th>Product Name</th>
                <th>Product Price</th>
                <th>Product Category</th>
              </tr>
              {state.cart.map((car) => (
                <tr>
                  <td>{car.title}</td>
                  <td>{car.price}</td>
                  <td>{car.category}</td>
                </tr>
              ))}
            </table>
            <hr />
            <div className="total">
              total price: {state.cart.reduce((acc, car) => acc + car.price, 0)}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}
