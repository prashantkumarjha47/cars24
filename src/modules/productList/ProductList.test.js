import renderer from "react-test-renderer";
import ProductList from "./ProductList";
import { ContextProvider } from "../../store/Context";

test("Filter component loaded", () => {
  const component = renderer.create(
    <ContextProvider>
      <ProductList />
    </ContextProvider>
  );
  expect(component.toJSON()).toMatchSnapshot();
});
