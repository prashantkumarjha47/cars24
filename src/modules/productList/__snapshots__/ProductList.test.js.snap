// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component loaded 1`] = `
<div
  className="product-list"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="product-list-left  col-12 col-sm-12 col-md-12 col-lg-4 col-xl-4"
      >
        <div
          className="filters"
        >
          <button
            aria-label={null}
            className="close-btn btn btn-link"
            onClick={[Function]}
            type="button"
          >
            <img
              alt="Close"
              src="close-icon.svg"
            />
          </button>
          <div
            className="container"
          >
            <form
              className=""
            >
              <div
                className="form-group"
              >
                <label
                  className=""
                  htmlFor="exampleSelect"
                >
                  Category
                </label>
                <select
                  className="form-control"
                  id="exampleSelect"
                  name="select"
                  onChange={[Function]}
                  value=""
                >
                  <option
                    value=""
                  >
                    
                  </option>
                  <option
                    value="Compact SUV"
                  >
                    Compact SUV
                  </option>
                  <option
                    value="Premium hatchback"
                  >
                    Premium hatchback
                  </option>
                  <option
                    value="Hatchback"
                  >
                    Hatchback
                  </option>
                </select>
              </div>
              <div
                className="form-group"
              >
                <label
                  className=""
                  htmlFor="exampleSelect"
                >
                  Price range
                </label>
                <div
                  aria-disabled={false}
                  className="input-range"
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onTouchStart={[Function]}
                >
                  <span
                    className="input-range__label input-range__label--min"
                  >
                    <span
                      className="input-range__label-container"
                    >
                      0
                    </span>
                  </span>
                  <div
                    className="input-range__track input-range__track--background"
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}
                  >
                    <div
                      className="input-range__track input-range__track--active"
                      style={
                        Object {
                          "left": "0%",
                          "width": "0%",
                        }
                      }
                    />
                    <span
                      className="input-range__slider-container"
                      style={
                        Object {
                          "left": "0%",
                          "position": "absolute",
                        }
                      }
                    >
                      <span
                        className="input-range__label input-range__label--value"
                      >
                        <span
                          className="input-range__label-container"
                        >
                          0
                        </span>
                      </span>
                      <div
                        aria-valuemax={20}
                        aria-valuemin={0}
                        aria-valuenow={0}
                        className="input-range__slider"
                        draggable="false"
                        onKeyDown={[Function]}
                        onMouseDown={[Function]}
                        onTouchStart={[Function]}
                        role="slider"
                        tabIndex="0"
                      />
                    </span>
                  </div>
                  <span
                    className="input-range__label input-range__label--max"
                  >
                    <span
                      className="input-range__label-container"
                    >
                      20
                    </span>
                  </span>
                </div>
              </div>
              <div
                className="form-group"
              >
                <label
                  className=""
                  htmlFor="exampleSelect"
                >
                  Vehicle Condition
                </label>
                <select
                  className="form-control"
                  id="exampleSelect"
                  name="select"
                  onChange={[Function]}
                  value=""
                >
                  <option
                    value="used"
                  >
                    used
                  </option>
                  <option
                    value="damaged"
                  >
                    damaged
                  </option>
                </select>
              </div>
              <div
                className="form-group"
              >
                <label
                  className=""
                  htmlFor="exampleSelect"
                >
                  Car make
                </label>
                <select
                  className="form-control"
                  id="exampleSelect"
                  name="select"
                  onChange={[Function]}
                  value=""
                >
                  <option
                    value="TATA"
                  >
                    TATA
                  </option>
                  <option
                    value="Kia"
                  >
                    Kia
                  </option>
                  <option
                    value="Hyundai"
                  >
                    Hyundai
                  </option>
                  <option
                    value="Ford"
                  >
                    Ford
                  </option>
                  <option
                    value="Maruti"
                  >
                    Maruti
                  </option>
                </select>
              </div>
              <div
                className="submit-row form-group"
              >
                <button
                  aria-label={null}
                  className="btn btn-secondary"
                  onClick={[Function]}
                  type="button"
                >
                  Apply
                </button>
                 
                <button
                  aria-label={null}
                  className="btn btn-secondary"
                  onClick={[Function]}
                  type="button"
                >
                  Reset
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
      <div
        className="product-list-right col-12 col-sm-12 col-md-12 col-lg-8 col-xl-8"
      >
        <div
          className="sorter-row"
        >
          <div
            className="sorter"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <div
                  className="search-row col-12 col-sm-12 col-md-12 col-lg-4"
                >
                  <input
                    className="form-control"
                    onChange={[Function]}
                    placeholder="Search"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="col-7 col-sm-8 col-md-8 col-lg-4"
                >
                  <select
                    className="form-control"
                    onChange={[Function]}
                    placeholder="Filter by price"
                    value=""
                  >
                    <option
                      value="ASC"
                    >
                      Low to High
                    </option>
                    <option
                      value="DESC"
                    >
                      High to Low
                    </option>
                  </select>
                </div>
                <div
                  className="filter-btn col-auto"
                >
                  <button
                    aria-label={null}
                    className="btn btn-info"
                    onClick={[Function]}
                    type="button"
                  >
                    Filter
                  </button>
                   
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="product-row"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="card-col col-12 col-sm-6 col-md-4 col-lg-4 col-xl-4"
              >
                <div
                  className="card"
                >
                  <img
                    alt="Card image cap"
                    className="card-img-top"
                    src="https://stimg.cardekho.com/images/car-images/large/Tata/Nexon/7297/1579769288048/224_Flame-Red_a52411.jpg"
                    width="100%"
                  />
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Tata Nexon
                    </h5>
                    <h6
                      className="mb-2 text-muted card-subtitle"
                    >
                      ₹ 600000
                    </h6>
                    <button
                      aria-label={null}
                      className="btn btn-secondary"
                      onClick={[Function]}
                    >
                      Book Now
                    </button>
                  </div>
                </div>
              </div>
              <div
                className="card-col col-12 col-sm-6 col-md-4 col-lg-4 col-xl-4"
              >
                <div
                  className="card"
                >
                  <img
                    alt="Card image cap"
                    className="card-img-top"
                    src="https://stimg.cardekho.com/images/car-images/large/Tata/Nexon/7297/1579769288048/224_Flame-Red_a52411.jpg"
                    width="100%"
                  />
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Kia Sonet
                    </h5>
                    <h6
                      className="mb-2 text-muted card-subtitle"
                    >
                      ₹ 700000
                    </h6>
                    <button
                      aria-label={null}
                      className="btn btn-secondary"
                      onClick={[Function]}
                    >
                      Book Now
                    </button>
                  </div>
                </div>
              </div>
              <div
                className="card-col col-12 col-sm-6 col-md-4 col-lg-4 col-xl-4"
              >
                <div
                  className="card"
                >
                  <img
                    alt="Card image cap"
                    className="card-img-top"
                    src="https://stimg.cardekho.com/images/car-images/large/Tata/Nexon/7297/1579769288048/224_Flame-Red_a52411.jpg"
                    width="100%"
                  />
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Ford EcoSport
                    </h5>
                    <h6
                      className="mb-2 text-muted card-subtitle"
                    >
                      ₹ 750000
                    </h6>
                    <button
                      aria-label={null}
                      className="btn btn-secondary"
                      onClick={[Function]}
                    >
                      Book Now
                    </button>
                  </div>
                </div>
              </div>
              <div
                className="card-col col-12 col-sm-6 col-md-4 col-lg-4 col-xl-4"
              >
                <div
                  className="card"
                >
                  <img
                    alt="Card image cap"
                    className="card-img-top"
                    src="https://stimg.cardekho.com/images/car-images/large/Tata/Nexon/7297/1579769288048/224_Flame-Red_a52411.jpg"
                    width="100%"
                  />
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Hyundai Venue
                    </h5>
                    <h6
                      className="mb-2 text-muted card-subtitle"
                    >
                      ₹ 675000
                    </h6>
                    <button
                      aria-label={null}
                      className="btn btn-secondary"
                      onClick={[Function]}
                    >
                      Book Now
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
